### DISCUSSION TITLE
bind fails for localhost on system without IPv6 DNS

### DISCUSSION THREAD

**Discussion by bertsky** [Kludex/uvicorn#1968] [Potential Issue] [answered]
Probably a regression from https://github.com/encode/uvicorn/pull/827:

If I do
```sh
echo 'from starlette.applications import Starlette; app = Starlette()' > server.py
uvicorn --host localhost --port 7000 server:app
```
that yields:
```
INFO:     Started server process [12]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
ERROR:    [Errno 99] error while attempting to bind on address ('::1', 7000, 0, 0): cannot assign requested address
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
```

The system in question does not resolve `localhost` to IPv6, only IPv4:
```sh
> host localhost
localhost.localdomain has address 127.0.0.1
> host -6 localhost
;; connection timed out; no servers could be reached
```

Version is `uvicorn 0.22.0 with CPython 3.11.3 on Linux`.

**Comment by asdorsey** [comment]
Also seeing this in a Docker container without IPv6.

Version is `uvicorn 0.24.0 with CPython 3.11.6 on Linux`.

**Reply by martin-schulze-e2m** [reply, marked as answer]
Still seems to be an issue in docker with `uvicorn 0.30.6 with CPython 3.11.6 on Linux`. We found that docker (pre v26.0) has a [bug regarding IPv6](https://github.com/docker/for-linux/issues/1374).
